rails-snow-queen/app/assets/javascripts/map.js

```diff
@@ -0,0 +1,110 @@
+class Map {
+	constructor(config) {
+    this.map = this.initMap();
+    this.drawingManager = this.initDrawingManager();
+    this.removeControl = this.initRemoveControl();
+    this.geocoder = this.initGeocoder();
+    this.marker = null;
+    this.geocodedAddress = null;
+    this.polygons = [];
+    this.onGeocodingResponse = config.onGeocodingResponse;
+    this.onPolygonsCreated = config.onPolygonsCreated;
+
+    this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
+    this.handleGeocodingResponse = this.handleGeocodingResponse.bind(this);
+    this.handlePolygonCreated = this.handlePolygonCreated.bind(this);
+    this.handleRemovePolygon = this.handleRemovePolygon.bind(this);
+    this.addListeners()
+  }
+  
+  initMap() {
+    return new google.maps.Map(document.getElementById('map'), {
+      zoom: 14,
+      center: {lat: 49.2860, lng: -122.8130},
+      mapTypeId: google.maps.MapTypeId.SATELLITE,
+      tilt:0
+    });
+  }
+  
+  initDrawingManager() {
+    const drawingOptions = {
+      fillColor: '#2DC1D6',
+      fillOpacity: 0.3,
+      strokeWeight: 5,
+      strokeColor: '#2DC1D6',
+      clickable: false,
+      // editable: true,
+      zIndex: 1
+    }
+    return new google.maps.drawing.DrawingManager({
+      drawingMode: google.maps.drawing.OverlayType.POLYGON,
+      drawingControl: true,
+      drawingControlOptions: {
+        position: google.maps.ControlPosition.TOP_CENTER,
+        drawingModes: ['polygon', 'rectangle']
+      },
+      map: this.map,
+      // markerOptions: {icon: 'https://developers.google.com/maps/documentation/javascript/examples/full/images/beachflag.png'},
+      // markerOptions: {icon: 'snow_flake.png'},
+      polygonOptions: drawingOptions,
+      rectangleOptions: drawingOptions
+    })
+  }
+  
+  initRemoveControl() {
+    const removeControlDiv = document.createElement('div');
+    removeControlDiv.classList.add("map-btn")
+    removeControlDiv.title = 'Click to remove selected area from the map';
+    removeControlDiv.innerHTML = 'Remove last';
+
+    removeControlDiv.index = 1;
```

Comment:
By the way, I [found out](https://developers.google.com/maps/documentation/javascript/controls#CustomPositioning) what this is for:

> The API places controls at each position by the order of an index property; controls with a lower index are placed first. For example, two custom controls at position BOTTOM_RIGHT will be laid out according to this index order, with lower index values taking precedence. By default, all custom controls are placed after placing any API default controls. You can override this behavior by setting a control's index property to be a negative value. Custom controls cannot be placed to the left of the logo or to the right of the copyrights.

Reply:
Good to know, not necessary to make any changes in this case right?

Reply:
nope, it was just because you had asked and I didn't have an answer